src/meshes/mesh_io.jl

```diff
@@ -269,7 +269,7 @@ function load_mesh_serial(mesh_file::AbstractString; n_cells_max, RealT)
         #
         # This should be replaced with something more robust and secure,
         # see https://github.com/trixi-framework/Trixi.jl/issues/541).
-        expr = Meta.parse(mapping_as_string)
+        expr = Meta.parseall(mapping_as_string)
```

Comment:
Why is this necessary? I am asking because `Meta.parse` is an entity documented and can thus be considered to be part of the public API of Julia, whereas `Meta.parseall` does not have a docstring

Reply:
In order to deal with inline comments, we join the lines with newlines instead of semicolons. `Meta.parse` parses only one expression, which is failing in this case. `parseall` can handle multiline code.

Reply:
OK, maybe you can add a comment in the code explaining this reason (or a link to this convo) for future readers, then it should be ok

Reply:
This version didn't work in some cases.
```julia
            mapping = @eval function (xi)
                $expr
                mapping(xi)
            end
```
This does not work if `expr` is a "toplevel" expression. I don't really know much about expressions, so I wasn't able to solve this problem.

Instead, I changed it to the following.
```julia
            mapping = eval(Meta.parse("""function (xi)
                $mapping_as_string
                mapping(xi)
            end
            """))
```
Here, we don't even need `parseall`, as the whole thing is just one expression. Well, unless someone passes code like
```julia
end
something()
begin
```
to the `StructuredMesh` as `mapping_as_string`, but then they can't really expect it to work, anyway.

Reply:
Thanks